File: core-examples/src/main/java/io/vertx/example/core/eventbus/pointtopoint/Sender.java
package io.vertx.example.core.eventbus.pointtopoint;

import io.vertx.core.AbstractVerticle;
import io.vertx.core.eventbus.EventBus;
import io.vertx.launcher.application.VertxApplication;

/*
 * @author <a href="http://tfox.org">Tim Fox</a>
 */
public class Sender extends AbstractVerticle {

  public static void main(String[] args) {
    VertxApplication.main(new String[]{Sender.class.getName(), "-cluster"});
  }

  @Override
  public void start() throws Exception {
    EventBus eb = vertx.eventBus();

    // Send a message every second

    vertx.setPeriodic(1000, v -> {

      eb.request("ping-address", "ping!")
        .onComplete(reply -> {
          if (reply.succeeded()) {
            System.out.println("Received reply " + reply.result().body());
          } else {
            System.out.println("No reply");
          }
        });

    });
  }
}

File: core-examples/src/main/kotlin/io/vertx/example/core/eventbus/pointtopoint/Sender.kt
package io.vertx.example.core.eventbus.pointtopoint


class Sender : io.vertx.core.AbstractVerticle()  {
  override fun start() {
    var eb = vertx.eventBus()

    // Send a message every second

    vertx.setPeriodic(1000, { v ->

      eb.request<Any>("ping-address", "ping!", { reply ->
        if (reply.succeeded()) {
          println("Received reply ${reply.result().body()}")
        } else {
          println("No reply")
        }
      })

    })
  }
}
